components/net_traits/image_cache.rs

```diff
@@ -572,29 +478,53 @@ impl ImageCache {
         })
     }
 
-    /// Return any available metadata or image for the given URL, or an indication that
-    /// the image is not yet available if it is in progress, or else reserve a slot in
-    /// the cache for the URL if the consumer can request images.
-    fn get_image_or_meta_if_available(&mut self,
-                                      url: ServoUrl,
-                                      placeholder: UsePlaceholder,
-                                      can_request: CanRequestImages)
-                                      -> Result<ImageOrMetadataAvailable, ImageState> {
-        if let Some(result) = self.get_completed_image_if_available(&url, placeholder) {
+    /// Handle a message from one of the decoder worker threads or from a sync
+    /// decoding operation.
+    fn handle_decoder(&self, msg: DecoderMsg) {
+        let image = match msg.image {
+            None => LoadResult::None,
+            Some(image) => LoadResult::Loaded(image),
+        };
+        self.complete_load(msg.key, image);
+    }
+
+    // Public API
+
+    pub fn new(webrender_api: webrender_traits::RenderApi) -> Arc<ImageCache> {
```

Comment:
Let's extract the public API into a trait that lives in net_traits/image_cache.rs, and move the implementation back into net_traits/image_cache.rs.